CMakeLists.txt

```
@@ -17,6 +17,14 @@ option(USE_XDG "Follow XDG-Basedir specification" ON) # Enabled by default
 
 # Absolute path reference for submodules
 set(WL_ROOT_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
+
+# Add custom includes to override system libs
+FILE(GLOB custom_include_directories auto_dependencies/*)
+foreach(custom_include_dir ${custom_include_directories})
+  message(STATUS "Using custom include directory '${custom_include_dir}'")
+  list(INSERT CMAKE_INCLUDE_PATH 0 "${custom_include_dir}")
```

```suggestion
  if(IS_DIRECTORY "${custom_include_dir}")
    message(STATUS "Using custom include directory '${custom_include_dir}'")
    list(INSERT CMAKE_INCLUDE_PATH 0 "${custom_include_dir}")
    include_directories(BEFORE SYSTEM "${custom_include_dir}")
  else()
    message(WARNING "Not a directory: '${custom_include_dir}' - Ignored.")
  endif()
```

Using `include_directories` fixes using auto_dependencies for me on Debian buster (oldstable). I think this should work with both old and new cmake...

I think it is best done right here, because you want these to be overrides anyway.

And the `IS_DIRECTORY` chack makes it safer.